### PR TITLE
DDS Security support

### PR DESCRIPTION
**Pull request by mikaelarguedas**
connects to ros2/ros2#339

### CODE REVIEW

File: rmw_connext_dynamic_cpp/src/functions.cpp [hunk 1]
```diff
@@ -198,9 +198,20 @@ rmw_init()
...
 
 rmw_node_t *
+rmw_create_secure_node(
+  const char * name,
+  const char * namespace_,
+  size_t domain_id,
+  const char * security_files_paths)
+{
+  (void)security_files_paths;
+  return create_node(rti_connext_dynamic_identifier, name, namespace_, domain_id, NULL);
```
Comment (dirk-thomas) [inline rmw_connext_dynamic_cpp/src/functions.cpp]: Shouldn't this return an error instead?

Comment (mikaelarguedas) [inline rmw_connext_dynamic_cpp/src/functions.cpp]: given that connext_dynamic is broken I don't know if this works or not so I didn't pass the variable. I'll pass it anyway given that the code path for connext and connext dynamic is exactly the same in the case of node creation. Thanks for pointing it out

File: rmw_connext_shared_cpp/src/shared_functions.cpp [hunk 2]
```diff
@@ -346,6 +347,103 @@ create_node(
     return NULL;
   }
 
+  if (security_options->security_root_path) {
```
Comment (dirk-thomas) [inline rmw_connext_shared_cpp/src/shared_functions.cpp]: You must check that `security_options` is not null before accessing its members.

Comment (mikaelarguedas) [inline rmw_connext_shared_cpp/src/shared_functions.cpp]: done in https://github.com/ros2/rmw_connext/pull/225/commits/dda0cd0b30fa03eb08e5e23ef4d7d6b5a4560198

**Pull request merged**